src/firstboot/firstboot.c

```diff
@@ -479,7 +479,7 @@ static int process_timezone(void) {
                                 return log_error_errno(r, "Failed to read host timezone: %m");
 
                         (void) mkdir_parents(etc_localtime, 0755);
-                        if (symlink(p, etc_localtime) < 0)
+                        if (symlink_atomic(p, etc_localtime) < 0)
                                 return log_error_errno(errno, "Failed to create %s symlink: %m", etc_localtime);
```

Comment:
Unlike `symlink()`, `symlink_atomic()` returns negative errno on error. Please pass the return value to the logger, like the following:
```
r = symlink_atomic(...);
if (r < 0)
        return log_error_errno(r, "...");
```